client/packages/system/src/Patient/DetailView/DetailView.tsx

```diff
@@ -171,8 +177,8 @@ export const PatientView: FC = () => {
 
   return (
     <React.Suspense fallback={<DetailViewSkeleton />}>
-      <ProgramDetailModal />
-      <EncounterDetailModal />
+      {current === PatientModal.Program ? <ProgramDetailModal /> : null}
```

Comment:
it's not everyone's favourite approach.. but I prefer having props

`<ProgramDetailModal visible={current === PatientModal.Program} />`

as it reads more nicely.
You can then return null in the modal component.

Is there are reason for not rendering the modals initially?

Reply:
It was a similar issue we had before, when leaving the modal do state was preserved, so when pressing cancel the modal closed but when reopening it again the old data discarded data was displayed. I need to unmount to re-fetch the initial data when reopening it again. (better way to do it?)

Reply:
hm. there are three levels of local state which is causing the actual problem. `useJsonForms` is holding on to the data in local state, as is the `useDocumentLoader` and then the rendering components all have state as well.

The unmounting is a solution - it just feels a little like brute force and I think we can solve the problem in a more react appropriate way. Looking into that now.

Reply:
I suggest ffbe5f386 - removing the `useState` from `useDocumentLoader` 
Any problems that I haven't noticed with doing that?